patsy/eval.py

```diff
@@ -29,7 +29,11 @@ def _all_future_flags():
     flags = 0
     for feature_name in __future__.all_feature_names:
         feature = getattr(__future__, feature_name)
-        if feature.getMandatoryRelease() > sys.version_info:
+        mr = feature.getMandatoryRelease()
+        # None means a planned feature was dropped, or at least postponed
+        # without a final decision; see, for example,
+        # https://docs.python.org/3.11/library/__future__.html#id2.
+        if mr is None or mr > sys.version_info:
```

Comment:
If is dropped, should the next line run?

Reply:
Answering that requires some reflection on what this code is trying to accomplish.

In this case, the PEP corresponding to the `annotations` feature was dropped in the sense that it was postponed twice, and a decision has not been made about whether it will be scheduled for a future release or rejected altogether. However, the feature is still available in `__future__`, and it might be scheduled for stabilization in the future.

I figured that this code is looking for future features that *could have been imported* and *are not included in the current Python release*. It seems like features with mandatory release `None` still satisfy this condition. It’s possible that I’m not seeing the full context, though.

Reply:
the patch maked patsy work on my python-3.11.0rc1 test

Reply:
From what I can see it looks like this code is trying to capture all possible flags that could be set by the user; so I think keeping the feature when `mr is None` makes sense.